app/controllers/orders_controller.rb

```diff
@@ -45,8 +47,14 @@ def get_all_proposals
     StudyClient::Node.where(cost_code: '!_none').all
   end
 
-  def item_option_selections
-    item.item_option_selections
+  def get_current_catalogues
+    Catalogue.where(current: true).all
+  end
+
+  def get_set_size
+    set_uuid = work_order.original_set_uuid
+    return 0 if set_uuid.nil?
+    SetClient::Set.find_with_materials(set_uuid).first.materials.length
```

Comment:
There is a `size` attribute in the meta information of Set if you wanted to save having to include fetching all the materials.